### DISCUSSION TITLE
Vertical engines

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#12604] [Ideas and suggestions]
Ballast tanks are cool and natural for submarines, but unnatural for drones

It's hard to make baby ballast tank maintainable (repair pump, kill flora (i know you can make it invulnerable))
It's weird that small drone can't overtake kastrull vertically because of speed cap
It's weird to see how 1 drop of water makes drone go from -18 to 36 descent speed
It's harder to control buoyancy on smaller tanks, i guess
It’s inconvenient that you can’t make drone’s vertical and horizontal speed equal (above 18)

So i suggest vertical engines

Maybe it's gonna be imba, but I would like to have such an option in sub editor

**Comment by Ek-Videogames** [comment]
I can see how this makes sense especially for battery-operated low-volume shuttle & drone craft.

Also could be an option on certain special larger submarines as an option for high-intensity situations (which is basically how i'd designed a modded sub using vertical engines) to either give better vertical maneuverability responsiveness or to temporarily offset loss of buoyancy due to flooding - albeit at the cost of a very high power usage.

Excerpt from steam discussions:
> Yeah I mean, engines as-is already consume ridiculous amounts of power. 
I feel like that alone wouldn't really cause any issue with vertical engines being overpowered.
As an option for using more power to augment vertical maneuverability, or as emergency lift to counteract flooding.

 > > Upwards speeds is capped around 15 while downwards speed is not
 
> As absolute psycho sub-designer/physicist I can confirm this. Submarine vertical acceleration is lower when 100% buoyant than it is when 0% buoyant, and because no submarine can ever fill more than 100% of it's volume with air - that means the terminal velocity for all submarines regardless of size is always ~15m/s. (While downwards terminal velocity is dependent on the vessel's ballast volume ratio)

**Comment by bookslayer10** [comment]
I find it's annoying how sub ascent velocity is always capped. Unlike horizontal speed that can be upgraded with engine power, you can't upgrade how much air you can fit in your sub. This idea would be good for drones, but I also think that vertical engines could be very fun and useful on standard submarines.

It would be great if this could be worked into a captain talent as well. Unlocking the talent unlocks a new wiring port on the submarine's engines that, when toggled, switches them from horizontal to vertical movement. Players would need to do their own wiring and hook up their own button to get this functionality, but it would let submarines rise and fall much faster than otherwise possible. There could even be intelligent designs that automatically turn on vertical engines when going straight upwards.

**Comment by SomeRandomNoobKekeke** [comment]
this should do it https://steamcommunity.com/sharedfiles/filedetails/?id=3087804415
